### DISCUSSION TITLE
Welcome to Fedora Remix for WSL Discussions!

### DISCUSSION THREAD

**Discussion by crramirez** [WhitewaterFoundry/Fedora-Remix-for-WSL#116] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by asemblR** [comment]
Iam getting 

"error: sqlite failure: CREATE TABLE IF NOT EXISTS 'Packages' (hnum INTEGER PRIMARY KEY AUTOINCREMENT,blob BLOB NOT NULL): locking protocol
error: cannot open Packages index using sqlite - No such file or directory (2)
error: cannot open Packages database in /var/lib/rpm
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'yum clean packages'.
Traceback (most recent call last):
  File "/usr/bin/yum", line 58, in <module>
    main.user_main(sys.argv[1:], exit_code=True)
  File "/usr/lib/python3.9/site-packages/dnf/cli/main.py", line 201, in user_main
    errcode = main(args)
  File "/usr/lib/python3.9/site-packages/dnf/cli/main.py", line 67, in main
    return _main(base, args, cli_class, option_parser_class)
  File "/usr/lib/python3.9/site-packages/dnf/cli/main.py", line 106, in _main
    return cli_run(cli, base)
  File "/usr/lib/python3.9/site-packages/dnf/cli/main.py", line 130, in cli_run
    ret = resolving(cli, base)
  File "/usr/lib/python3.9/site-packages/dnf/cli/main.py", line 176, in resolving
    base.do_transaction(display=displays)
  File "/usr/lib/python3.9/site-packages/dnf/cli/cli.py", line 238, in do_transaction
    self.gpgsigcheck(install_pkgs)
  File "/usr/lib/python3.9/site-packages/dnf/cli/cli.py", line 278, in gpgsigcheck
    result, errmsg = self._sig_check_pkg(po)
  File "/usr/lib/python3.9/site-packages/dnf/base.py", line 1297, in _sig_check_pkg
    sigresult = dnf.rpm.miscutils.checkSig(ts, po.localPkg())
  File "/usr/lib/python3.9/site-packages/dnf/rpm/miscutils.py", line 90, in checkSig
    raise ValueError('Unexpected return value %r from hdr.sprintf when checking signature.' % siginfo)
ValueError: Unexpected return value 'RSA/SHA256, Tue 26 Jan 2021 07:44:10 AM PST, Key ID 1161ae6945719a39' from hdr.sprintf when checking signature.

On a fresh install of fedoraremix on WIndows 10 21N1
I tried replacing the db with the "rpmrebuilddb.114"   db in the directory above the RPM  dir.

Thank YOU!

**Reply by crramirez** [reply]
Hello,

An update of dnf or its dependencies is doing something that is unsupported on WSL1. Please run upgrade.sh it has a workaround inside.

Regards,
Carlos

**Comment by PavelSosin-320** [comment]
Thanks a lot! I thought it is a part of the Fedora remix to 34 upgrade.

**Reply by crramirez** [reply]
WSL1 is getting more and more difficult to maintain. Soon the only distros that will be able to run in it will be enterprise ones RHEL, and so on and old LTS versions. It emulates the kernel 4.4 and these newer distros all are based on 5.x

**Comment by BazTheLinuxGuy** [comment]
Is it possible to use X, such as the Xorg server, in Fedora Remix in WSL?

**Reply by crramirez** [reply]
Yes of course, just install the one that you prefer and Fedora Remix will use it automatically

**Reply by BazTheLinuxGuy** [reply]
There must be a lot of dependencies, just installing Xorg and a few associated programs didn't do the trick. Is there any way I can find the dependencies to run X Windows, say, with Xfce4?

**Reply by crramirez** [reply]
Hello,

OK I didn't understand what you meant. You want to run a full desktop.

First you need to install an XServer for Windows like Vcxsrv or X410. Then follow these instructions : https://www.whitewaterfoundry.com/blog/2020/4/10/installing-mate-in-fedora-remix-for-wsl

Regards,
Carlos
